doc/sampling.md

```diff
@@ -0,0 +1,113 @@
+
+- `usersvc` requests whose operation name is `healthcheck` are never sampled.
+- Other `usersvc` requests are sampled 50% of the time.
+- `authsvc` requests are sampling 100% of the time.
+- All other requests are sampled 10% of the time.
+
+Note that this behavior is relevant only if the tracer being configured is the
+_first_ in the trace.
+
+When a trace is created, its root span is evaluated against each sampling rule
+in order.  The first rule that matches determines the probability that the
+trace will be sampled.  If no rule matches, then the trace is subject to the
+sampling rates governed by the Datadog Agent, as explained above.
+
+Sampling rules can be configured programmatically in `std::string
+TracerOptions::sampling_rules` or via the environment variable
+`DD_TRACE_SAMPLING_RULES`.  In either case, the rules are expressed as a JSON
+array of objects.  Each object supports the following properties:
+```
+[{
+    "service": <the root span's service name, or any if absent>,
+    "name": <the root span's operation name, or any if absent>,
+    "sample_rate": <the probability of sampling the trace, or 1.0 if absent>
+}, ...]
+```
+
+`DD_TRACE_SAMPLE_RATE`
+----------------------
+Setting a (numeric) value for the `DD_TRACE_SAMPLE_RATE` environment variable
+effectively appends a sampling rule to the tracer's array of sampling rules:
+```
+[
+    ...,
+    {"sample_rate": $DD_TRACE_SAMPLE_RATE
+]
+```
+Now there is a sampling rule that matches _any_ trace, and so traces that do
+not match an earlier sampling rule are subject to the configured sampling rate.
+
+Note that using `DD_TRACE_SAMPLE_RATE` means that the Datadog Agent no longer
+governs the sampling rate of any traces produced by the tracer.  The implicit
+"catch-all" rule, with the configured sampling rate, always takes precedence
+over the Agent-based fallback.
+
+`double TracerOptions::sample_rate`
+-----------------------------------
+This configuration option has the same meaning as the `DD_TRACE_SAMPLE_RATE`
+environment variable.  Note that the environment variable overrides the
+`TracerOptions` field if both are specified. 
+
+`DD_TRACE_RATE_LIMIT`
+---------------------
+Sampling rules (and, by extension, `DD_TRACE_SAMPLE_RATE`) specify the
+_probability_ that a trace will be sampled, but they do not specify the maximum
+number of traces that may be produced in a given time period.
+
+`DD_TRACE_RATE_LIMIT` is the maximum number of traces, per second, that may be
+sampled by the tracer on account of sampling rules or `DD_TRACE_SAMPLE_RATE`.
+The limit applies globally across all applicable traces, i.e. there is not a
+separate limit for each sampling rule.
+
+`DD_TRACE_RATE_LIMIT` is a floating point number, but is usually specified as an integer, e.g.
+```shell
+export DD_TRACE_RATE_LIMIT=200
+```
+for a limit of 200 traces per second.
+
+If this limit is not configured, its default value is 100 traces per second.
+
```

Comment:
Is it worth mentioning anything about rates being "between 0.0 and 1.0" for people that aren't familiar with probabilities?
Also the behavior of the limits when there's multiple threads or worker processes?
And the fallback to priority sampling / agent-provided rates?

Reply:
> Is it worth mentioning anything about rates being "between 0.0 and 1.0" for people that aren't familiar with probabilities?

Yes, I will mention that.

> Also the behavior of the limits when there's multiple threads or worker processes?

😅 In the interest of airing out our laundry, yes, I will add a section for that.

> And the fallback to priority sampling / agent-provided rates?

I think that's mentioned, but I will emphasize it more (or mention it if I'm mistaken).